luigi/contrib/bigquery_avro.py

```diff
@@ -86,10 +86,64 @@ def get_fields_with_description(bq_fields, avro_fields):
             new_fields = []
             for field in bq_fields:
                 avro_field = avro_fields[field[u'name']]
+                field_type = type(avro_field.type)
+
+                #Primitive and top level complex type support
                 field[u'description'] = avro_field.doc
-                if field[u'type'] == u'RECORD' and hasattr(avro_field.type, 'fields_dict'):
-                    field[u'fields'] = \
-                        get_fields_with_description(field[u'fields'], avro_field.type.fields_dict)
+
+                #Enum Support
+                if field_type is avro.schema.EnumSchema:
+                    field[u'description'] = avro_field.type.doc
```

Comment:
Does this and the following assignments overwrite the `field[u'description'] = avro_field.doc` above intentionally? I wonder if the type documentation should really take precedence like that. The field documentation is presumably more specific to the concrete usage and the dataset we're loading, thus more interesting to the user than the type documentation, isn't it? Conceivably both could be useful... Concatenate? :)

Reply:
Cleaned up the confusion considerably here.  Realistically grabbing the data from "avro_field.doc" only works in 3 scenarios:  PrimitiveSchema, MapSchema, and PrimitiveSchema nested in a UnionSchema.  To make the code more explicit I moved this call directly to where it was needed.  And just for clarity all other times the call returns 'None' which is automatically dropped so the field would populate no data to the Big Query description